papermario/src/code_80850_len_3060.c

```diff
@@ -108,15 +128,56 @@ INCLUDE_ASM(code_80850_len_3060, sync_status_menu);
 
 INCLUDE_ASM(code_80850_len_3060, reset_status_menu);
 
+// uses a jumptable, which we need .rodata support for.
 INCLUDE_ASM(code_80850_len_3060, is_ability_active);
 
 s32 is_partner_ability_active(void) {
     return 0;
 }
 
-INCLUDE_ASM(code_80850_len_3060, add_coins);
+s16 add_coins(s32 amt) {
+    player_data *playerData = &gPlayerData;
+    s16 temp_v0;
+
+    temp_v0 = playerData->coins + amt;
+    playerData->coins = temp_v0;
+    if (temp_v0 >= 1000) {
+        playerData->coins = 999;
+    }
+    if (playerData->coins < 0) {
+        playerData->coins = 0;
+    }
 
-INCLUDE_ASM(code_80850_len_3060, add_star_points);
+    if (amt > 0) {
+        u32 temp_v0_2 = playerData->totalCoinsEarned + amt;
+        playerData->totalCoinsEarned = temp_v0_2;
+        if (temp_v0_2 > 99999) {
+            playerData->totalCoinsEarned = 99999;
+        }
+    }
+    return playerData->coins;
+}
+
+s8 add_star_points(s32 amt) {
+    player_data *playerData = &gPlayerData;
+    player_data *playerData2 = &gPlayerData;
+    s8 temp_v0;
+
+    temp_v0 = playerData->starPoints + amt;
+
+    //TODO: probably a macro!
+    playerData2->starPoints = temp_v0;
+    if (temp_v0 >= 0x65) {
```

Comment:
this should be able to just be `> 0x64` and also I prefer to use decimal for numbers that represent things the player can see and understand